plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestHiveS3MinioQueries.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.HiveQueryRunner;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.metastore.HiveMetastoreConfig;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DataProviders;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.containers.Minio;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.containers.Minio.MINIO_ACCESS_KEY;
+import static io.trino.testing.containers.Minio.MINIO_SECRET_KEY;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHiveS3MinioQueries
+        extends AbstractTestQueryFramework
+{
+    private Minio minio;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        minio = closeAfterClass(Minio.builder().build());
+        minio.start();
+
+        return HiveQueryRunner.builder()
+                .setMetastore(queryRunner -> {
+                    File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("hive_data").toFile();
+                    return new FileHiveMetastore(
+                            new NodeVersion("testversion"),
+                            HDFS_ENVIRONMENT,
+                            new HiveMetastoreConfig().isHideDeltaLakeTables(),
+                            new FileHiveMetastoreConfig()
+                                    .setCatalogDirectory(baseDir.toURI().toString())
+                                    .setDisableLocationChecks(true) // matches Glue behavior
```

Comment:
why not directly test with Glue + HMS since this test actually depends on metastore behaviour (before your change).

Reply:
Glue doesn't normalize. 
Glue tests will be easier to add after https://github.com/trinodb/trino/pull/17804.

Yes, we could ALSO test with HMS, but after this change the behavior cannot really be metastore-dependent, can it?
i am not opposed to such tests, but with apple m1 it's hard to write them :(